### PR TITLE
r/sagemaker_notebook_instance: Root access to Sagemaker notebook instance

### PR DESCRIPTION
**Pull request by dinakar29**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #NA

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
Ability to configure root access for Sagemaker notebook instances
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSSagemakerNotebookInstance_root_access'

==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSSagemakerNotebookInstance_root_access -timeout 120m
=== RUN   TestAccAWSSagemakerNotebookInstance_root_access
=== PAUSE TestAccAWSSagemakerNotebookInstance_root_access
=== CONT  TestAccAWSSagemakerNotebookInstance_root_access
--- PASS: TestAccAWSSagemakerNotebookInstance_root_access (652.27s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	653.837s
```

### CODE REVIEW

File: aws/resource_aws_sagemaker_notebook_instance.go [hunk 1]
```diff
@@ -74,6 +74,17 @@ func resourceAwsSagemakerNotebookInstance() *schema.Resource {
 				ForceNew: true,
 			},
 
+			"root_access": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  sagemaker.RootAccessEnabled,
+				ValidateFunc: validation.StringInSlice([]string{
+					sagemaker.RootAccessDisabled,
```
Comment (DrFaust92) [inline aws/resource_aws_sagemaker_notebook_instance.go:83]: nit, can you change this to `sagemaker.RootAccess_Values()`

File: aws/resource_aws_sagemaker_notebook_instance_test.go [hunk 2]
```diff
@@ -544,6 +587,33 @@ data "aws_iam_policy_document" "assume_role" {
 `, notebookName, notebookName)
 }
 
+func testAccAWSSagemakerNotebookInstanceConfigRootAccess(notebookName string, rootAccess string) string {
+	return fmt.Sprintf(`
+resource "aws_sagemaker_notebook_instance" "foo" {
+	name = %[1]q
+	role_arn = "${aws_iam_role.foo.arn}"
```
Comment (DrFaust92) [inline aws/resource_aws_sagemaker_notebook_instance_test.go:594]: use tf12 syntax

File: aws/resource_aws_sagemaker_notebook_instance_test.go [hunk 3]
```diff
@@ -544,6 +587,33 @@ data "aws_iam_policy_document" "assume_role" {
...
+	instance_type = "ml.t2.medium"
+	root_access = %[2]q
+}
+
+resource "aws_iam_role" "foo" {
+	name = %[1]q
+	path = "/"
+	assume_role_policy = "${data.aws_iam_policy_document.assume_role.json}"
```
Comment (DrFaust92) [inline aws/resource_aws_sagemaker_notebook_instance_test.go:602]: same as above

**Review by DrFaust92 — CHANGES_REQUESTED** [review]
Hey, small change request.

**Review by DrFaust92 — APPROVED** [review]
LGTM

```
--- PASS: TestAccAWSSagemakerNotebookInstance_root_access (699.92s)
```

**Comment by dinakar29** [comment]
@DrFaust92 - Is this good to merge? We are eagerly anticipating this being available in a near future release.

**Review by ewbankkit — APPROVED** [review]
LGTM.

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSSagemakerNotebookInstance_root_access\|TestAccAWSSagemakerNotebookInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSSagemakerNotebookInstance_root_access\|TestAccAWSSagemakerNotebookInstance_basic -timeout 120m
=== RUN   TestAccAWSSagemakerNotebookInstance_basic
=== PAUSE TestAccAWSSagemakerNotebookInstance_basic
=== RUN   TestAccAWSSagemakerNotebookInstance_root_access
=== PAUSE TestAccAWSSagemakerNotebookInstance_root_access
=== CONT  TestAccAWSSagemakerNotebookInstance_basic
=== CONT  TestAccAWSSagemakerNotebookInstance_root_access
--- PASS: TestAccAWSSagemakerNotebookInstance_basic (305.93s)
--- PASS: TestAccAWSSagemakerNotebookInstance_root_access (625.81s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	625.849s
```

**Comment by breathingdust** [comment]
LGTM! 🚀  Thanks @dinakar29 

Verified Acceptance Tests in Commercial (us-west-2)

```console
make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSSagemakerNotebookInstance_root_access\|TestAccAWSSagemakerNotebookInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSSagemakerNotebookInstance_root_access\|TestAccAWSSagemakerNotebookInstance_basic -timeout 120m
=== RUN   TestAccAWSSagemakerNotebookInstance_basic
=== PAUSE TestAccAWSSagemakerNotebookInstance_basic
=== RUN   TestAccAWSSagemakerNotebookInstance_root_access
=== PAUSE TestAccAWSSagemakerNotebookInstance_root_access
=== CONT  TestAccAWSSagemakerNotebookInstance_basic
=== CONT  TestAccAWSSagemakerNotebookInstance_root_access
--- PASS: TestAccAWSSagemakerNotebookInstance_basic (332.52s)
--- PASS: TestAccAWSSagemakerNotebookInstance_root_access (671.71s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	673.305s
```

Verified Acceptance Tests in GovCloud (us-gov-west-1)

```console
make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSSagemakerNotebookInstance_root_access\|TestAccAWSSagemakerNotebookInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSSagemakerNotebookInstance_root_access\|TestAccAWSSagemakerNotebookInstance_basic -timeout 120m
=== RUN   TestAccAWSSagemakerNotebookInstance_basic
=== PAUSE TestAccAWSSagemakerNotebookInstance_basic
=== RUN   TestAccAWSSagemakerNotebookInstance_root_access
=== PAUSE TestAccAWSSagemakerNotebookInstance_root_access
=== CONT  TestAccAWSSagemakerNotebookInstance_basic
=== CONT  TestAccAWSSagemakerNotebookInstance_root_access
--- PASS: TestAccAWSSagemakerNotebookInstance_basic (297.88s)
--- PASS: TestAccAWSSagemakerNotebookInstance_root_access (693.07s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	694.530s
```

**Pull request merged**